lightning/src/util/events.rs

```diff
@@ -196,6 +196,94 @@ impl_writeable_tlv_based_enum_upgradable!(HTLCDestination,
+#[derive(Clone, Debug)]
+pub struct AnchorDescriptor {
+	/// A unique identifier used along with `channel_value_satoshis` to re-derive the
+	/// [`InMemorySigner`] required to sign `input`.
+	///
+	/// [`InMemorySigner`]: crate::chain::keysinterface::InMemorySigner
+	pub channel_keys_id: [u8; 32],
+	/// The value in satoshis of the channel we're attempting to spend the anchor output of. This is
+	/// used along with `channel_keys_id` to re-derive the [`InMemorySigner`] required to sign
+	/// `input`.
+	///
+	/// [`InMemorySigner`]: crate::chain::keysinterface::InMemorySigner
+	pub channel_value_satoshis: u64,
+	/// The transaction input's outpoint corresponding to the commitment transaction's anchor
+	/// output.
+	pub outpoint: OutPoint,
+}
+
+/// Describes a pending HTLC in a commitment transaction.
+#[derive(Clone, Debug)]
+pub struct CommitmentPendingHTLC {
+	/// Whether the pending HTLC originated from LDK.
+	pub outbound: bool,
+	/// The amount, in satoshis, of the pending HTLC's output in the commitment transaction.
+	pub amount_sat: u64,
+	/// The absolute height in the chain at which the pending HTLC expires and can be claimed back
+	/// by its originator.
+	pub expiry: u32,
+	/// The index of the pending HTLC's output in the commitment transaction.
+	pub output_index: u32,
+}
+
+/// Represents the different types of transactions, originating from LDK, to be bumped.
+#[derive(Clone, Debug)]
+pub enum BumpTransaction {
+	/// Indicates that a channel featuring anchors outputs is to be closed by broadcasting the local
+	/// commitment transaction. Since commitment transactions have a static feerate pre-agreed upon,
+	/// they may need additional fees to be attached through a child transaction using the popular
+	/// [Child-Pays-For-Parent](https://bitcoinops.org/en/topics/cpfp) fee bumping technique. This
+	/// child transaction must include the anchor input described within `anchor_descriptor` along
+	/// with additional inputs to meet the target feerate. Failure to meet the target feerate
+	/// decreases the confirmation odds of the transaction package (which includes the commitment
+	/// and child anchor transactions), possibly resulting in a loss of funds. Once the transaction
+	/// is constructed, it must be fully signed for and broadcasted by the consumer of the event
+	/// along with the `commitment_tx` enclosed. Note that the `commitment_tx` must always be
+	/// broadcast first, as the child anchor transaction depends on it.
+	///
+	/// The consumer should be able to sign for any of the additional inputs included within the
+	/// child anchor transaction. To sign its anchor input, an [`InMemorySigner`] should be
+	/// re-derived through [`KeysManager::derive_channel_keys`] with the help of
+	/// [`AnchorDescriptor::channel_keys_id`] and [`AnchorDescriptor::channel_value_satoshis`].
+	///
+	/// It is possible to receive more than an instance of this event if a valid child anchor
```

Comment:
nit: maybe this is a bit clearer

```suggestion
	/// It is possible to receive more than one instance of this event if a valid child anchor
```